core/src/main/java/com/datastrato/gravitino/catalog/CapabilityHelpers.java

```diff
@@ -199,11 +199,13 @@ private static Transform applyCapabilities(Transform transform, Capability capab
 
     } else if (transform instanceof Transforms.ListTransform) {
       return Transforms.list(
-          applyCapabilities(((Transforms.ListTransform) transform).fieldNames(), capabilities));
+          applyCapabilities(((Transforms.ListTransform) transform).fieldNames(), capabilities),
+          ((Transforms.ListTransform) transform).assignments());
```

Comment:
You also need to apply `CapabilityHelpers.applyCaseSensitive` to assignments partitions' names

Reply:
I found `applyCaseSensitive` for partition is defined in `com.datastrato.gravitino.catalog.PartitionNormalizeDispatcher`, should we move it into `CapabilityHelpers`?